src/openrct2/core/Json.cpp

```diff
@@ -26,42 +25,120 @@ namespace Json
-        fileData[fileLength] = '\0';
+        auto fileData = std::string(static_cast<size_t>(fileLength) + 1, '\0');
+        fs.Read(static_cast<void*>(fileData.data()), fileLength);
 
-        json_error_t jsonLoadError;
-        json = json_loads(fileData, 0, &jsonLoadError);
-        Memory::Free(fileData);
+        json_t json;
 
-        if (json == nullptr)
+        try
         {
-            throw JsonException(&jsonLoadError);
+            json = json_t::parse(fileData, nullptr, true, true);
+        }
+        catch (const json_t::exception& e)
+        {
+            throw JsonException(String::Format("Unable to parse JSON file (%s)\n\t%s", path, e.what()));
         }
 
         return json;
     }
 
-    void WriteToFile(const utf8* path, const json_t* json, size_t flags)
+    void WriteToFile(const utf8* path, const json_t& jsonData, int indentSize)
     {
         // Serialise JSON
-        const char* jsonOutput = json_dumps(json, flags);
+        std::string jsonOutput = jsonData.dump(indentSize);
 
         // Write to file
         auto fs = OpenRCT2::FileStream(path, OpenRCT2::FILE_MODE_WRITE);
-        size_t jsonOutputSize = String::SizeOf(jsonOutput);
-        fs.Write(jsonOutput, jsonOutputSize);
+        fs.Write(jsonOutput.data(), jsonOutput.size());
     }
 
-    json_t* FromString(std::string_view raw)
+    json_t FromString(std::string_view raw)
     {
-        json_t* root;
-        json_error_t error;
-        root = json_loadb(raw.data(), raw.size(), 0, &error);
-        if (root == nullptr)
+        json_t json;
+
+        try
+        {
+            json = json_t::parse(raw, nullptr, true, true);
+        }
+        catch (const json_t::exception& e)
+        {
+            log_error("Unable to parse JSON string (%s)\n\t%s", raw, e.what());
+        }
+
+        return json;
+    }
+
+    json_t FromVector(const std::vector<uint8_t>& vec)
+    {
+        json_t json;
+
+        try
         {
-            throw JsonException(&error);
+            json = json_t::parse(vec.begin(), vec.end());
         }
-        return root;
+        catch (const json_t::exception& e)
+        {
+            log_error("Unable to parse JSON vector (%s)\n\t%s", vec.data(), e.what());
+        }
+
+        return json;
+    }
+
+    std::string GetString(const json_t& jsonObj, const std::string& defaultValue)
+    {
+        if (jsonObj.is_string())
+        {
+            return jsonObj.get<std::string>();
+        }
+        else
+        {
+            return defaultValue;
+        }
+    }
+
+    bool GetBoolean(const json_t& jsonObj, bool defaultValue)
+    {
+        if (jsonObj.is_boolean())
+        {
+            return jsonObj.get<bool>();
+        }
+        else
+        {
+            return defaultValue;
+        }
+    }
+
+    json_t AsObject(const json_t& jsonObj)
+    {
+        if (jsonObj.is_object())
+        {
+            return jsonObj;
+        }
+
+        return json_t::object();
+    }
```

Comment:
```suggestion
    std::string GetString(const json_t& jsonObj, const std::string& defaultValue)
    {
        return jsonObj.is_string() ? jsonObj.get<std::string>() : defaultValue;
    }

    bool GetBoolean(const json_t& jsonObj, bool defaultValue)
    {
        return jsonObj.is_boolean() ? jsonObj.get<bool>() : defaultValue;
    }

    json_t AsObject(const json_t& jsonObj)
    {
        return jsonObj.is_object() ? jsonObj : json_t::object();
    }
```

I'd go for ternaries in here, but I know some of the other guys in the team don't love it, so you don't need to do anything in here if you don't want ;P

Reply:
I don't hate ternaries per se, but I am far more cautious because they get hard to read very quickly.

In this case, though, I think using ternaries is fine - I think they actually improve legibility here.

Reply:
+1 from me for ternaries here.